packages/functions/src/utils.ts

```diff
@@ -332,3 +340,105 @@ export const validateJSON = <T extends Record<string, any>>(
   return [null, constructValidationErrors(validate.errors || [], prefix), {}];
 };
 // #endregion JSONValidation
+
+// #region CalculateInterval
+
+/**
+ * @param {string | null} interval - String slot interval
+ * Converts a string slot interval to a number e.g:
+ * `null => 0`;
+ * `"21:00 - 21:20" => 0.5`;
+ * `"16:00 - 17:00" => 1.0`;
+ * `"22:00 - 23:30" => 1.5`;
+ * `"22:00 - 24:00" => 2`;
+ * `"22:20 - 24:00" => 2`;
+ */
+export const calculateIntervalDuration = (interval: string | null) => {
```

Comment:
We already have logic to calculate an interval duration. Duplicating this piece of logic enables some bugs (misalignment of what's seen on one screen and what's seen on another one). We can easily avoid them by keeping a single implementation of this funciton throughout the codebase.

I suggest to factor out [the existing function](https://github.com/eisbuk/EisBuk/blob/master/packages/ui/src/AttendanceSheet/utils.ts#L11C14-L32) into the [shared package](https://github.com/eisbuk/EisBuk/blob/master/packages/shared/src/utils/slots.ts) and import it from there into both `functions` and `ui`. A little change will be needed, since the current function returns a string for display, but here we need a value to calculate a grand total, but the core logic needs to be kept in a single place, and not duplicated.